rmf_traffic/test/unit/test_Profile.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <rmf_traffic/Profile.hpp>
+#include <rmf_traffic/DetectConflict.hpp>
+#include <rmf_traffic/geometry/Circle.hpp>
+#include <rmf_traffic/geometry/Box.hpp>
+
+#include <rmf_utils/catch.hpp>
+
+using namespace std::chrono_literals;
+
+SCENARIO("Testing Construction")
+{
+  using Profile = rmf_traffic::Profile;
+
+  WHEN("Vicinity is not passed into the constructor")
+  {
+    const auto circle_1 = rmf_traffic::geometry::make_final_convex<
+        rmf_traffic::geometry::Circle>(1.0);
+    const auto circle_2 = rmf_traffic::geometry::make_final_convex<
+        rmf_traffic::geometry::Circle>(2.0);
+
+    auto profile = Profile{circle_1};
+
+    auto& footprint = profile.footprint();
+    auto& vicinity = profile.vicinity();
+
+    CHECK(footprint == circle_1);
+    CHECK(vicinity == circle_1);
+    CHECK(footprint->get_characteristic_length() == Approx(1.0));
+    CHECK(vicinity->get_characteristic_length() == Approx(1.0));
+
+    WHEN("Only footprint is updated")
+    {
+      profile.footprint(circle_2);
+      CHECK(footprint == circle_2);
+      CHECK(profile.footprint()->get_characteristic_length() == Approx(2.0));
+      // profile.vicinity() should return footprint
```

Comment:
```suggestion
      // profile.vicinity() should return footprint
      CHECK(profile.vicinity() == profile.footprint());
```

Let's make that an explicit test instead of just a comment.